File: packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiConnectCodes.java
package com.alternadom.wifiiot;

/**
 * Unified Wi-Fi connect result codes; values must match {@code lib/wifi_connect_codes.dart} and
 * iOS {@code WifiConnectCodes.swift}. Empty string means success.
 */
public final class WifiConnectCodes {
  private WifiConnectCodes() {}

  public static final String OK = "";

  public static final String INVALID_SSID = "invalid_ssid";

  public static final String INVALID_BSSID = "invalid_bssid";
  public static final String WEP_NOT_SUPPORTED = "wep_not_supported";
  public static final String UNKNOWN = "unknown";

  public static final String NETWORK_CONFIGURATION_FAILED = "network_configuration_failed";
  public static final String DISCONNECT_FAILED = "disconnect_failed";
  public static final String ENABLE_NETWORK_FAILED = "enable_network_failed";
  public static final String CONNECTION_TIMEOUT = "connection_timeout";
  public static final String CONNECTION_UNAVAILABLE = "connection_unavailable";

  public static final String NETWORK_SUGGESTION_INTERNAL = "network_suggestion_internal";
  public static final String NETWORK_SUGGESTION_APP_DISALLOWED = "network_suggestion_app_disallowed";
  public static final String NETWORK_SUGGESTION_DUPLICATE = "network_suggestion_duplicate";
  public static final String NETWORK_SUGGESTION_EXCEEDS_MAX_PER_APP =
      "network_suggestion_exceeds_max_per_app";
  public static final String NETWORK_SUGGESTION_REMOVE_INVALID = "network_suggestion_remove_invalid";
  public static final String NETWORK_SUGGESTION_ADD_NOT_ALLOWED = "network_suggestion_add_not_allowed";
  public static final String NETWORK_SUGGESTION_INVALID = "network_suggestion_invalid";
  public static final String NETWORK_SUGGESTION_RESTRICTED_BY_ADMIN =
      "network_suggestion_restricted_by_admin";
  public static final String NETWORK_SUGGESTION_FAILED = "network_suggestion_failed";
}

File: packages/wifi_iot/ios/Classes/WifiConnectCodes.swift
import Foundation
import NetworkExtension

/// Unified Wi-Fi connect result codes; values must match `lib/wifi_connect_codes.dart` and Android `WifiConnectCodes.java`. Empty string means success.
enum WifiConnectCodes {
    static let ok = ""
    static let invalidSsid = "invalid_ssid"
    static let invalidBssid = "invalid_bssid"
    static let wepNotSupported = "wep_not_supported"
    static let unknown = "unknown"

    static let networkConfigurationFailed = "network_configuration_failed"
    static let disconnectFailed = "disconnect_failed"
    static let enableNetworkFailed = "enable_network_failed"
    static let connectionTimeout = "connection_timeout"
    static let connectionUnavailable = "connection_unavailable"

    // Android addNetworkSuggestions statuses (same strings for cross-platform handling in Dart)
    static let networkSuggestionInternal = "network_suggestion_internal"
    static let networkSuggestionAppDisallowed = "network_suggestion_app_disallowed"
    static let networkSuggestionDuplicate = "network_suggestion_duplicate"
    static let networkSuggestionExceedsMaxPerApp = "network_suggestion_exceeds_max_per_app"
    static let networkSuggestionRemoveInvalid = "network_suggestion_remove_invalid"
    static let networkSuggestionAddNotAllowed = "network_suggestion_add_not_allowed"
    static let networkSuggestionInvalid = "network_suggestion_invalid"
    static let networkSuggestionRestrictedByAdmin = "network_suggestion_restricted_by_admin"
    static let networkSuggestionFailed = "network_suggestion_failed"

    static let invalidConfiguration = "invalid_configuration"
    static let invalidSsidNative = "invalid_ssid_native"
    static let invalidWpaPassphrase = "invalid_wpa_passphrase"
    static let invalidWepPassphrase = "invalid_wep_passphrase"
    static let invalidEapSettings = "invalid_eap_settings"
    static let invalidHs20Settings = "invalid_hs20_settings"
    static let internalError = "internal_error"
    static let operationPending = "operation_pending"
    static let systemConfigurationError = "system_configuration_error"
    static let unknownError = "unknown_error"
    static let joinOnceNotSupported = "join_once_not_supported"
    static let notInForeground = "not_in_foreground"
    static let userDenied = "user_denied"

    static let postApplySsidUnavailable = "post_apply_ssid_unavailable"
    static let postApplySsidMismatch = "post_apply_ssid_mismatch"

    static let iosVersionUnsupported = "ios_version_unsupported"
    static let pluginInternal = "plugin_internal"

    @available(iOS 11.0, *)
    static func from(neError: NEHotspotConfigurationError) -> String {
        switch neError {
        case .invalid:
            return invalidConfiguration
        case .invalidSSID:
            return invalidSsidNative
        case .invalidWPAPassphrase:
            return invalidWpaPassphrase
        case .invalidWEPPassphrase:
            return invalidWepPassphrase
        case .invalidEAPSettings:
            return invalidEapSettings
        case .invalidHS20Settings:
            return invalidHs20Settings
        case .internal:
            return internalError
        case .pending:
            return operationPending
        case .systemConfiguration:
            return systemConfigurationError
        case .unknown:
            return unknownError
        case .joinOnceNotSupported:
            return joinOnceNotSupported
        case .alreadyAssociated:
            return ok
        case .applicationIsNotInForeground:
            return notInForeground
        case .userDenied:
            return userDenied
        default:
            return unknownError
        }
    }
}
